subsys/dfu/src/dfu_target.c

```diff
@@ -68,10 +68,16 @@ int dfu_target_init(int img_type, size_t file_size)
 	}
 
 	/* The user is re-initializing with an previously aborted target.
-	 * Avoid re-initializing to ensure that the download can continue where
-	 * it left off.
+	 * Avoid re-initializing generally to ensure that the download can
+	 * continue where it left off.
 	 */
 	if (new_target == current_target) {
+		if (img_type == DFU_TARGET_IMAGE_TYPE_MODEM_DELTA) {
+			/* Re-initializing is required for modem upgrades to
+			 * re-open the DFU socket that is closed on abort.
+			 */
+			return current_target->init(file_size);
+		}
```

Comment:
Let's do it like this instead:

```
	if (new_target == current_target &&
		img_type != DFU_TARGET_IMAGE_TYPE_MODEM_DELTA) {
		return 0;
	}

	current_target = new_target;

	return current_target->init(file_size);

```

then it's more clear to the user that we are not accidentallly forgetting to update current_target.

Reply:
Changed, tested and appended the comment with regards to modem upgrades above.